bundles/admin/admin-announcements/instance.js

```diff
@@ -1,42 +1,94 @@
-        conf.name = 'admin-announcements';
-        conf.flyoutClazz = 'Oskari.admin.bundle.admin-announcements.Flyout';
-        this.popupControls = null;
-    }, {
-
-        afterStart: function () {
-        },
-
-        showEditPopup: function (controller, announcement) {
+Oskari.clazz.defineES('Oskari.admin.admin-announcements.instance',
+    class AdminAnnouncements extends BasicBundle {
+        constructor () {
+            super();
+            this.__name = 'admin-announcements';
+            this.loc = Oskari.getMsg.bind(null, this.__name);
+            this.popupControls = null;
+            this.service = null;
+        }
+
+        _startImpl () {
+            this.service = this.sandbox.getService('Oskari.framework.announcements.service.AnnouncementsService');
+            this.setupAdminTooling();
+        }
+
+        showEditPopup (id) {
             if (this.popupControls) {
                 this.popupCleanup();
             }
+            const announcement = this.service.getAnnouncement(id);
             const onClose = () => this.popupCleanup();
-            this.popupControls = showEditPopup(controller, announcement, onClose);
-        },
-        popupCleanup: function () {
+            const onSubmit = (announcement) => {
+                const isNew = !announcement.id;
+                this.service.submitAnnouncement(announcement, err => this.notifySubmit(err, isNew));
+                onClose();
+            };
+            const onDelete = () => {
+                this.service.deleteAnnouncement(id, err => this.notifyDelete(err));
+                onClose();
+            };
+            this.popupControls = showEditPopup(announcement, onSubmit, onDelete, onClose);
+        }
+
+        notifyDelete (error) {
+            if (error) {
+                Messaging.error(this.loc('messages.deleteFailed'));
+            } else {
+                Messaging.success(this.loc('messages.deleteSuccess'));
+            }
+        }
+
+        notifySubmit (error, isNew) {
+            const key = isNew ? 'save' : 'update';
+            if (error) {
+                Messaging.error(this.loc(`messages.${key}Failed`));
+            } else {
+                Messaging.success(this.loc(`messages.${key}Success`));
+            }
+        }
+
+        popupCleanup () {
             if (this.popupControls) {
                 this.popupControls.close();
             }
             this.popupControls = null;
         }
 
-    }, {
-        'extend': ['Oskari.userinterface.extension.DefaultExtension']
+        setupAdminTooling () {
+            if (!this.service) {
+                return;
+            }
+            const editTool = new Tool('announcements-edit', EditOutlined);
+            editTool.setTooltip(this.loc('tools.edit'));
+            editTool.setTypes(['announcement']);
+            editTool.setCallback((id) => this.showEditPopup(id));
+            this.service.addTool(editTool);
```

Comment:
The Tool class should be provided by announcement bundle since it's the one that uses it/needs to understand it's usage. I would refactor this to just admin bundle service.addTool('edit', tooltip, ['announcement'], callback). Then we wouldn't have to import the tool class here.